DataCollector/DoctrineDataCollector.php

```diff
@@ -136,9 +136,33 @@ public function collect(Request $request, Response $response, \Exception $except
             }
         }
 
+        $groupedQueries = array();
+        foreach ($this->data['queries'] as $connection => $queries) {
```

Comment:
I'm wondering whether we really need to perform this extra work at collection time (which adds time to the page being loaded). It only computes something based purely on already collected data (and not needed by the toolbar, only by the panel), so we could do it on demand in ``getGroupedQueries`` (and ensure that the template calls it only once and then keeps it in a variable)